base/string.jl

```diff
@@ -968,8 +968,8 @@ byte_string_classify(s::ByteString) = byte_string_classify(s.data)
     # 1: valid ASCII
     # 2: valid UTF-8
 
-is_valid_ascii(s::Union(Array{UInt8,1},ByteString)) = byte_string_classify(s) == 1
-is_valid_utf8(s::Union(Array{UInt8,1},ByteString)) = byte_string_classify(s) != 0
+isvalid(::Type{ASCIIString}, s::Union(Array{UInt8,1},ByteString)) = byte_string_classify(s) == 1
```

Comment:
I never said that I was the only one who cares about string-handling performance, a number of people have already told me that they were very happy to see me trying to do something about it...
My point has been all along that a *lot* of people do care about the string-handling performance, which is why some attention should be paid to it... (I've also said, a number of times, that current poor performance wasn't because the Julia team did a bad job, but rather, it just hasn't been the focus of their time... the places where they *have* spent their time, is pretty darned remarkable, IMO).
I'm well aware of said trade-offs, I've spent quite a long time developing and maintaining a quite large code base... if you (or anybody) can point me to a better way of writing those conversion functions, so that Julia takes care of spitting out the specialized cases, that *doesn't* kill the performance gains, then I'm all for it.  #11004 was my very first Julia code that wasn't just wrappers, and I've already got some ideas about how I might be able to condense it (although the bug in Julia with inference might have to get fixed first, and I have no idea [yet] how to accomplish that).  The code is very simple, and has more inline comments than what was there before... so I don't understand the worries about maintainability.

About adding more specialized methods, is there any way of telling just where the old code might still be getting used?  There is so much in Base, that it seems like quite a lot might already be going directly to the old code, no matter what I do, not just some Win32 calls.